R/epidist.R

```diff
@@ -352,21 +353,28 @@ validate_epidist <- function(epidist) {
       ) %in%
         attributes(epidist)$names,
     "epidist must contain a disease (single character string)" =
-      is.character(epidist$disease$disease) &&
-        length(epidist$disease$disease) == 1,
+    checkmate::test_character(epidist$disease$disease, len = 1),
     "epidist must contain an epidemiological distribution" =
-      is.character(epidist$epi_dist) && length(epidist$epi_dist) == 1,
+      checkmate::test_character(epidist$epi_dist, len = 1),
     "epidist must contain a <distribution> or <distcrete> distribution or NA" =
-      inherits(epidist$prob_dist, c("distribution", "distcrete")) ||
-        is.na(epidist$prob_dist) || is.character(epidist$prob_dist),
+    checkmate::test_multi_class(
+      epidist$prob_dist, classes = c("distribution", "distcrete")
+    ) || checkmate::test_character(
+      epidist$prob_dist, any.missing = TRUE, len = 1
+    ),
```

Comment:
Suggest `test_string()` instead of specifying `len = 1` - {checkmate} has all these specific options it's hard to remember them all.

Reply:
Resolved in 065e1ba9834e238f3f6e8afa8015872a644911cb.